packages/app-desktop/gui/Navigator.jsx

```diff
@@ -9,10 +9,18 @@ class NavigatorComponent extends Component {
 		if (newProps.route) {
 			const screenInfo = this.props.screens[newProps.route.routeName];
 			const devMarker = Setting.value('env') === 'dev' ? ' (DEV)' : '';
-			const windowTitle = [`Joplin${devMarker}`];
+			const windowTitle = [];
 			if (screenInfo.title) {
 				windowTitle.push(screenInfo.title());
+			} else if (newProps.route.routeName == 'Main' && newProps.currentFolder && newProps.currentNote) {
+				const folderTitle = newProps.currentFolder.title.trim();
+				let noteTitle = newProps.currentNote.title.trim();
```

Comment:
How does it handle long titles?

Reply:
I'm not sure what you meant by this. Can you explain bit more? I guess you meant to truncate/ellipsis long titles to fit the titlebar

Reply:
Yes I meant what to do if the title is longer than a certain limit. A note title can have any size, even thousands of characters so we don't want to crash the app or have display issues in that case. So indeed, perhaps you could set a limit of x characters and add ellipsis after that.